### PR TITLE
fix(css-in-js) Keep newlines in CSS-in-JS Templates (Fixes: #5147)

### PR DESCRIPTION
**Pull request by onurtemizkan**
Fixes: #5147 

Keep newlines after CSS-in-JS quasi-literal substitutes. 

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: tests/multiparser_js_css/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -158,4 +188,35 @@ styled.div\`
...
+\`;
+
+styled.div\`
+  \${foo} bar;
+\`;
+
+styled.span\`
+  \${foo} \${bar}
+  baz: \${foo}
+\`;
```
Comment (evilebottnawi) [inline tests/multiparser_js_css/__snapshots__/jsfmt.spec.js.snap]: Let's add test with semicolon on end:
```js
styled.span`
${foo};
${bar};
`
```

```js
styled.span`
${foo}: ${bar};
`
```

```js
styled.span`
${foo}: ${bar}
`
```

And with newline(s) between `:` and `;`.

Comment (onurtemizkan) [reply tests/multiparser_js_css/__snapshots__/jsfmt.spec.js.snap]: Updated :+1:

File: tests/multiparser_js_css/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -158,4 +211,56 @@ styled.div\`
...
+styled.span\`
+  \${foo}: \${bar};
+\`;
+
+styled.span\`
+  \${foo}: \${bar};
+\`;
+
+styled.span\`
+  \${foo}: \${bar};
```
Comment (evilebottnawi) [inline tests/multiparser_js_css/__snapshots__/jsfmt.spec.js.snap]: Looks we insert unnecessary semicolon, need fix it

File: tests/multiparser_js_css/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -158,4 +211,56 @@ styled.div\`
...
+  \${foo};
+  \${bar};
+\`;
+
+styled.span\`
+  \${foo}: \${bar};
+\`;
+
+styled.span\`
+  \${foo}: \${bar};
```
Comment (evilebottnawi) [inline tests/multiparser_js_css/__snapshots__/jsfmt.spec.js.snap]: Also semicolon was inserted :disappointed:

**Comment by onurtemizkan** [comment]
@evilebottnawi updated 😊

**Comment by evilebottnawi** [comment]
@onurtemizkan Looks i find one strange problem, maybe not related, maybe related, will be great if you investigate.

Input:
```js
export const FlexBox = styled.div`
  ${$var}

  > * {
      width: ${p => (p.howManyPerRow ? `${100 / p.howManyPerRow}%` : 'auto')};
  }
`
```

Output:
```js
export const FlexBox = styled.div`
  ${$var}

  > * {
    width: ${p => (p.howManyPerRow ? `${100 / p.howManyPerRow}%` : "auto")};
  }
`;
```

All fine. But:

Input:
```js
export const FlexBox = styled.div`
    ${$var}

    > * {
        width: ${p => (p.howManyPerRow ? `${100 / p.howManyPerRow}%` : 'auto')};
    }
`
```

Output:
```js
export const FlexBox = styled.div`
  ${$var}

    > * {
    width: ${p => (p.howManyPerRow ? `${100 / p.howManyPerRow}%` : "auto")};
  }
`;
```

`> * ` have extra indent

**Comment by onurtemizkan** [comment]
@evilebottnawi Sure, checking.

**Comment by lydell** [comment]
@onurtemizkan Did you find anything? Would be awesome to get this into the next release. I guess we could merge as-is, too.

**Comment by onurtemizkan** [comment]
@lydell, I haven't found it yet. Let's merge this and I'll create another ticket to continue if it's ok.

**Comment by lipis** [comment]
Shall we take this in..?

**Comment by evilebottnawi** [comment]
@lipis feel free to merge, problem with indent should be solved in other PR

**Pull request merged**